cmd/collector/main.go

```diff
@@ -70,11 +74,21 @@ func main() {
 	server := thrift.NewServer(ch)
 	server.Register(jc.NewTChanCollectorServer(jaegerBatchesHandler))
 	server.Register(zc.NewTChanZipkinCollectorServer(zipkinSpansHandler))
+
 	portStr := ":" + strconv.Itoa(*builder.CollectorPort)
 	listener, err := net.Listen("tcp", portStr)
 	if err != nil {
 		logger.Fatal("Unable to start listening on channel", zap.Error(err))
 	}
 	ch.Serve(listener)
-	select {}
+
+	r := mux.NewRouter()
```

Comment:
>We already use mux elsewhere

We do, but not in the collector.

>yarpc doesn't have nearly the same market penetration

I don't think we should exclude libraries from consideration because they don't have market penetration, especially when they could fit our use case well. 

While `mux` works for now, what's the plan when we want to support a different protocol (say gRPC)? With yarpc, this would be a very simple change that consistently applies the same error handling across transports. 

I feel that we seriously evaluate yarpc before settling on a solution. 

@yurishkuro What's your opinion on this?